template/Makefile

```diff
@@ -45,6 +45,11 @@ dev: ## Install development dependencies
 	npm install
 	opam install --deps-only --with-test --with-doc -y .
 
+.PHONY: create_switch
+create_switch:
+  # Create a local opam switch
+	opam switch create . 4.10.0 --deps-only --with-test
```

Comment:
I think you'll want to use `--no-install` when creating the switch, because some of the dependencies need to be pinned first.